src/commands/BeautifyCommand.ts

```diff
@@ -19,6 +19,10 @@ export class BeautifyCommand extends BaseCommand {
         this.readConfig(programArgs),
         this.readText(filePath),
       ]).then(([config, text]) => {
+        if (!text) {
+          const error = new Error("No text in the file or stdin.");
```

Comment:
For many languages, it's valid to have a blank source file. In my opinion we should think of some other check. One thing we could check is that stdin is not a TTY if we are reading code from there. That situation is much more likely to happen accidentally rather than deliberately. In principle, a user could wish to type (or copy-paste) source code straight into a terminal window and have it beautified, but I don't think that's very common.

Reply:
A change like that should be made to [readFromStdin()](https://github.com/Unibeautify/unibeautify-cli/blob/d33d46a61ff693984b4965770ea0b53ab2673395/src/commands/BeautifyCommand.ts#L91-L112).  I want to give the user some sort of feedback if there is nothing for Unibeautify to do.

Reply:
OK. I think there are two separate cases here:
* We're told to read code from stdin, but stdin is a TTY -> This is an error.
* We're told to read code from a file/pipe, but that file/pipe is empty -> This is not an error, but we can display a warning message (for the CLI, the message would go to stderr).

Reply:
Probably the warning should only be shown if code is read from stdin. If an editor plugin is implemented using the CLI, and the editor is set to beautify code on save, then many users will be saving blank (freshly made) source files and the editor plugin will show a warning about the empty file in some popup/status window.

Likewise, if you type `unibeautify foo.js` in the shell and `foo.js` is empty, is that something the user should be warned about?

`generate-some-code | unibeautify` would be a more likely source of errors if `generate-some-code` does something unexpected?

Reply:
Just realized most editor plugins will be reading code from stdin too. Hmm. Then we'd need a command line flag to disable the warning. Maybe just a field inside `--config-json="{...}"`

Reply:
@Glavin001 thoughts?

Reply:
> Likewise, if you type `unibeautify foo.js` in the shell and foo.js is empty, is that something the user should be warned about?

and

> For many languages, it's valid to have a blank source file. In my opinion we should think of some other check. 

I think an empty string can be valid source code. I don't see a need to notify the user their own code is empty.
If this becomes a feature request by users then we can consider it, however, at this time I'd prefer not to.